### PR TITLE
Add support for Bearer Auth Tokens

### PR DESCRIPTION
**Pull request by nateprewitt**
This PR will introduce functionality to support Bearer auth tokens via a new TokenProviderChain. This will supply auth info via the `Authorization` header with the Bearer prefix as specified in [RFC6750](https://datatracker.ietf.org/doc/html/rfc6750#section-2.1).

### CODE REVIEW

**Review by dlm6693 — APPROVED** [review]
Mostly just a bunch of nits and questions. No blockers on shipping this 🚀

File: botocore/tokens.py [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+            self._frozen_token = self._refresh_using()
+        except Exception:
+            logger.warning(
+                "Refreshing token failed during the %s refresh period.",
+                refresh_type,
+                exc_info=True,
+            )
+            if refresh_type == "mandatory":
+                # This refresh was mandatory, error must be propagated back
+                raise
```
[the remark this answers is not in the document]
Comment (nateprewitt) [reply botocore/tokens.py:121]: I don't think this is a terminal state if another provider can satisfy the requirement.

File: botocore/tokens.py [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+                f"configuration does not exist."
+            )
+            raise InvalidConfigError(error_msg=error_msg)
+
+        missing_configs = []
+        for var in self._SSO_CONFIG_VARS:
+            if var not in sso_config:
+                missing_configs.append(var)
+
+        if missing_configs:
```
[the remark this answers is not in the document]
Comment (nateprewitt) [reply botocore/tokens.py:227]: Is it possible for a list to be truthy with 0 elements?

File: botocore/tokens.py [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+                f'The profile "{profile_name}" is configured to use the SSO '
+                f'token provider but the "{sso_session_name}" sso_session '
+                f"configuration does not exist."
+            )
+            raise InvalidConfigError(error_msg=error_msg)
+
+        missing_configs = []
+        for var in self._SSO_CONFIG_VARS:
+            if var not in sso_config:
+                missing_configs.append(var)
```
[the remark this answers is not in the document]
Comment (nateprewitt) [reply botocore/tokens.py:225]: There's other logic that needs to be added here later. We've left this structure in place to simplify the follow up PR.

File: botocore/tokens.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (dlm6693) [reply botocore/tokens.py:227]: No, but I've received feedback from you in the past to be more explicit with truthy statements specifically on iterables to check length

**Pull request merged**